src/devices/Mlx90640/Mlx90640.cs

```diff
@@ -0,0 +1,1072 @@
+            }
+
+            for (int i = 0; i < 768; i++)
+            {
+                temp = (float)(alphaTemp[i] * Math.Pow(2, alphaScale));
+                _alpha[i] = (ushort)(temp + 0.5);
+
+            }
+
+            _alphaScale = alphaScale;
+        }
+
+        private void ExtractOffsetParameters()
+        {
+            int[] occRow = new int[24];
+            int[] occColumn = new int[32];
+            int p = 0;
+            short offsetRef;
+            byte occRowScale;
+            byte occColumnScale;
+            byte occRemScale;
+
+            occRemScale = (byte)(_eeData[16] & 0x000F);
+            occColumnScale = (byte)((_eeData[16] & 0x00F0) >> 4);
+            occRowScale = (byte)((_eeData[16] & 0x0F00) >> 8);
+            offsetRef = (short)(_eeData[17]);
+            if (offsetRef > 32767)
+            {
+                offsetRef = (short)(offsetRef - 65536);
+            }
+
+            for (int i = 0; i < 6; i++)
+            {
+                p = i * 4;
+                occRow[p + 0] = (_eeData[18 + i] & 0x000F);
+                occRow[p + 1] = (_eeData[18 + i] & 0x00F0) >> 4;
+                occRow[p + 2] = (_eeData[18 + i] & 0x0F00) >> 8;
+                occRow[p + 3] = (_eeData[18 + i] & 0xF000) >> 12;
+            }
+
+            for (int i = 0; i < 24; i++)
+            {
+                if (occRow[i] > 7)
+                {
+                    occRow[i] = occRow[i] - 16;
+                }
+            }
+
+            for (int i = 0; i < 8; i++)
+            {
+                p = i * 4;
+                occColumn[p + 0] = (_eeData[24 + i] & 0x000F);
+                occColumn[p + 1] = (_eeData[24 + i] & 0x00F0) >> 4;
+                occColumn[p + 2] = (_eeData[24 + i] & 0x0F00) >> 8;
+                occColumn[p + 3] = (_eeData[24 + i] & 0xF000) >> 12;
+            }
+
+            for (int i = 0; i < 32; i++)
+            {
+                if (occColumn[i] > 7)
+                {
+                    occColumn[i] = occColumn[i] - 16;
+                }
+            }
+
+            for (int i = 0; i < 24; i++)
+            {
+                for (int j = 0; j < 32; j++)
+                {
+                    p = 32 * i + j;
+                    _offset[p] = (_eeData[64 + p] & 0xFC00) >> 10;
+                    if (_offset[p] > 31)
+                    {
+                        _offset[p] = _offset[p] - 64;
+                    }
+
+                    _offset[p] = _offset[p] * (1 << occRemScale);
+                    _offset[p] = (offsetRef + (occRow[i] << occRowScale) + (occColumn[j] << occColumnScale) + _offset[p]);
+                }
+            }
+        }
+
+        private void ExtractKtaPixelParameters()
+        {
+            int p = 0;
+            sbyte[] ktaRC = new sbyte[4];
+            sbyte ktaRoCo;
+            sbyte ktaRoCe;
+            sbyte ktaReCo;
+            sbyte ktaReCe;
+            byte ktaScale1;
+            byte ktaScale2;
+            byte split;
```

Comment:
general comment: you should declare variables as close to where they're actually used instead of C-like code...